### PR TITLE
check root-dir

### PR DESCRIPTION
**Pull request by datamel**
Use of `root-dir` in a rose-suite.conf raises a warning. 
This PR also fixes the flake8 issues that were causing master to fail, and fixes mypy failure too with adding `--install-types` to the GH Actions mypy command.

To test `cylc install` a workflow with a rose-suite.conf containing root-dir config. This logic existed before but was skipping e.g.
 ```
root-dir{share/cycle}=$TMPDIR

### CODE REVIEW

File: cylc/rose/entry_points.py [hunk 1]
```diff
@@ -103,13 +103,20 @@ def get_rose_vars(srcdir=None, opts=None):
...
-
-    # Warn if root-dir set in config:
-    if 'root-dir' in config_tree.node:
-        LOG.warning(
-            'You have set "root-dir", which at Cylc 8 does nothing. '
-            'See Cylc Install documentation.'
-        )
+    # Warn if root-dir set in config
+    old_rose_suite_run_config = (
+        config_tree.node.get_value(["rose-suite-run"]))
```
Comment (oliver-sanders) [inline cylc/rose/entry_points.py:108]: Just to confuse you the `root-dir` setting goes in a `rose-suite-run` section in the site-user config but *not* in the suite config.

* http://metomi.github.io/rose/doc/html/api/configuration/site-and-user.html#rose:conf:rose.conf[rose-suite-run]root-dir
* http://metomi.github.io/rose/doc/html/api/configuration/suite.html#rose:conf:rose-suite.conf|root-dir

Comment (datamel) [reply cylc/rose/entry_points.py:108]: Ah doh. I'll change this. Want me to change this to check the site-user config or not check?

Comment (oliver-sanders) [reply cylc/rose/entry_points.py:108]: I think we can ignore the site-user config.

File: setup.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 }
 TESTS_REQUIRE = [
     'coverage>=5.0.0',
-    'flake8',
+    'flake8-*',
```
Comment (oliver-sanders) [inline setup.py:36]: I don't think you can glob pypi modules, try replacing this with:

```
    'flake8-broken-line>=0.3.0',
    'flake8-bugbear>=21.0.0',
    'flake8-builtins>=1.5.0',
    'flake8-comprehensions>=3.5.0',
    'flake8-debugger>=4.0.0',
    'flake8-mutable>=1.2.0',
```

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**